component-tools/src/main/java/org/talend/sdk/component/tools/exec/CarMain.java

```diff
@@ -390,8 +391,8 @@ private static File jarFromResource(final ClassLoader loader, final String resou
             if ("jar".equals(url.getProtocol())) {
                 final String spec = url.getFile();
                 final int separator = spec.indexOf('!');
-                return new File(decode(new URL(spec.substring(0, separator)).getFile()));
-
+                return new File(
+                        URLDecoder.decode(new URL(spec.substring(0, separator)).getFile(), "UTF-8"));
```

Comment:
We can use the constant here instead of string "UTF-8"
java.nio.charset.StandardCharsets#UTF_8
That method has an overload with param as Charset.
Same comment for all invocations of this method

Reply:
Using StandardCharsets isn't java 8 compliant.

Reply:
Why? The class has existed since 1.7

Reply:
the method `decode(String, Charset)` was introduced with jdk10.